File: sinain-chat-agent/smoke_client.py
"""Tiny WS client to exercise the sidecar (stands in for sinain-core's ChatService).
Usage: python smoke_client.py "your message"   (sidecar must be running)"""
import asyncio, json, sys, time
import websockets


async def main() -> None:
    msg = sys.argv[1] if len(sys.argv) > 1 else "In one sentence, what is a vector embedding?"
    port = 9610
    async with websockets.connect(f"ws://127.0.0.1:{port}") as ws:
        t0 = time.perf_counter(); ttfc = None
        await ws.send(json.dumps({"message": msg, "context": {"kind": "main", "seed": ""}}))
        async for raw in ws:
            ev = json.loads(raw)
            t = (time.perf_counter() - t0) * 1000
            if ev["type"] == "token":
                if ttfc is None:
                    ttfc = t; print(f"  [ttfc {ttfc:.0f}ms]", flush=True)
                print(ev["text"], end="", flush=True)
            elif ev["type"] == "tool_call":
                print(f"\n  → tool: {ev['tool_name']}({ev.get('tool_args')})", flush=True)
            elif ev["type"] == "tool_result":
                print(f"  ← {ev['tool_result'][:80]!r}", flush=True)
            elif ev["type"] == "done":
                print(f"\n  [done {t:.0f}ms]", flush=True); break
            elif ev["type"] == "error":
                print(f"\n  [ERROR] {ev['text']}", flush=True); break


if __name__ == "__main__":
    asyncio.run(main())

File: sinain-chat-agent-js/src/smoke-client.ts
/** Smoke client — mirrors sinain-chat-agent/smoke_client.py.
 *
 *   npm run smoke -- "what do you know about my recent work?"
 *   npm run smoke -- --status            (health probe only)
 *   npm run smoke -- --cancel-after 2 "long question"   (test mid-turn cancel)
 */
import { WebSocket } from "ws";

const args = process.argv.slice(2);
const statusOnly = args.includes("--status");
const cancelIdx = args.indexOf("--cancel-after");
const cancelAfterS = cancelIdx >= 0 ? Number(args[cancelIdx + 1]) : 0;
const message = args
  .filter((a, i) => !a.startsWith("--") && (cancelIdx < 0 || i !== cancelIdx + 1))
  .join(" ") || "reply with exactly: pong";
const port = process.env.SINAIN_CHAT_WS_PORT || "9610";

const ws = new WebSocket(`ws://127.0.0.1:${port}`);
const t0 = Date.now();
let firstToken = 0;

ws.on("open", () => {
  if (statusOnly) {
    ws.send(JSON.stringify({ type: "status" }));
    return;
  }
  ws.send(JSON.stringify({ message, context: { kind: "main" } }));
  if (cancelAfterS > 0) {
    setTimeout(() => {
      console.log(`\n[smoke] sending cancel after ${cancelAfterS}s`);
      ws.send(JSON.stringify({ cancel: true }));
    }, cancelAfterS * 1000);
  }
});

ws.on("message", (raw) => {
  const ev = JSON.parse(String(raw)) as Record<string, unknown>;
  switch (ev.type) {
    case "status":
      console.log(`[smoke] status: ${JSON.stringify(ev)}`);
      ws.close();
      break;
    case "token":
      if (!firstToken) {
        firstToken = Date.now();
        console.log(`[smoke] first token @ ${((firstToken - t0) / 1000).toFixed(2)}s`);
      }
      process.stdout.write(String(ev.text ?? ""));
      break;
    case "tool_call":
      console.log(`\n[smoke] tool_call ${ev.tool_name} ${JSON.stringify(ev.tool_args)}`);
      break;
    case "tool_result":
      console.log(`[smoke] tool_result ${ev.tool_name}: ${String(ev.tool_result).slice(0, 120)}…`);
      break;
    case "usage_tick":
      console.log(`\n[smoke] usage_tick ${JSON.stringify(ev.usage)}`);
      break;
    case "done":
      console.log(`\n[smoke] done @ ${((Date.now() - t0) / 1000).toFixed(2)}s usage=${JSON.stringify(ev.usage)}`);
      ws.close();
      break;
    case "error":
      console.error(`\n[smoke] ERROR: ${ev.text} usage=${JSON.stringify(ev.usage)}`);
      ws.close();
      break;
  }
});

ws.on("error", (e) => {
  console.error(`[smoke] ws error: ${e.message}`);
  process.exit(1);
});
ws.on("close", () => process.exit(0));
